### DISCUSSION TITLE
make a global export method

### DISCUSSION THREAD

**Discussion by 350030173** [vfsfitvnm/frida-il2cpp-bridge#64] [Q&A]
How do I make a global export method that allows me to quickly input instructions, such as: hook_my_class(" unityEngine.ui "," unityEngine.ui.text ")

![1](https://user-images.githubusercontent.com/20263746/145832491-7f7f697a-38ec-4e6c-bbb5-ecb747fe66ff.png)
![2](https://user-images.githubusercontent.com/20263746/145832512-5767067a-fc42-442c-a362-ee3e18bb0766.png)

**Comment by vfsfitvnm** [comment]
Just declare it as global:

```ts
Il2Cpp.perform(() => {
    (global as any).test_fn = function () {
        console.log("Global:", Unity.version);
    }
});
```

You may use `globalThis` instead of `global` it depends on your typescript workplace.

PS: you don't need to use `assemblies` and then `filterClasses`, it's slow, do this instead:
```ts
Il2Cpp.trace().classes(myClass).commit().detailed().build();
```

**Reply by 350030173** [reply]
wow,thank you very much,now ，Now I don't have to re-attach every time

**Reply by vfsfitvnm** [reply]
You actually don't have to attach every time. If you build the script in watch mode frida automatically re-launch the script when you save it (e.g, make a change, press ctrl + s, script is now executed again)
